Modules/_testinternalcapi.c

```diff
@@ -35,6 +36,38 @@ get_recursion_depth(PyObject *self, PyObject *Py_UNUSED(args))
 }
 
 
+static PyObject*
+codecs_unregister(PyObject *self, PyObject *args)
+{
+    PyObject *search_function;
+
+    if (!PyArg_ParseTuple(args, "O:codecs_unregister",
+                          &search_function)) {
+        return NULL;
+    }
+    if (search_function == NULL) {
+        return NULL;
+    }
+
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    if (interp->codec_search_path == NULL) {
+        return NULL;
+    }
+
+    Py_ssize_t n = PyList_Size(interp->codec_search_path);
+    PyObject *list = PyList_New(0);
+    for (Py_ssize_t i = 0; i < n; i++) {
+        PyObject *item = PyList_GetItem(interp->codec_search_path, i);
```

Comment:
I'm surprised that list.remove() is not available as a C function.

Maybe PyList_Remove() should be added? The problem is the O(n) cost of this function, but the Python list.remove(x) has the same cost.

Reply:
> I'm surprised that list.remove() is not available as a C function.

Agree. Adding the basic function is more important than the O(n) cost ;)

Reply:
And I will create another bpo to do this job.

Reply:
we have a conclusion in https://bugs.python.org/issue41871.